pkg/api/targetgroup_deployment.go

```diff
@@ -0,0 +1,83 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/common-fate/apikit/apio"
+	"github.com/common-fate/common-fate/pkg/storage"
+	"github.com/common-fate/common-fate/pkg/types"
+	"github.com/common-fate/ddb"
+)
+
+// Your GET endpoint
+// (GET /api/v1/target-group-deployments)
+func (a *API) ListTargetGroupDeployments(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	res := types.ListTargetGroupDeploymentsResponse{
+		Res: []types.TargetGroupDeployment{},
+	}
+
+	listTargetGroupDeployments := storage.ListTargetGroupDeployments{}
+
+	_, err := a.DB.Query(ctx, &listTargetGroupDeployments)
+
+	if err != nil && err != ddb.ErrNoItems {
+		apio.Error(ctx, w, err)
+		return
+	}
+
+	if len(listTargetGroupDeployments.Result) == 0 {
+		res.Res = []types.TargetGroupDeployment{}
+	}
+	for _, r := range listTargetGroupDeployments.Result {
+		res.Res = append(res.Res, r.ToAPI())
+	}
+
+	apio.JSON(ctx, w, res, http.StatusOK)
+}
+
+// (POST /api/v1/target-group-deployments)
+func (a *API) CreateTargetGroupDeployment(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+
+	var b types.CreateTargetGroupDeploymentRequest
+	err := apio.DecodeJSONBody(w, r, &b)
+	if err != nil {
+		apio.Error(ctx, w, err)
+		return
+	}
+
+	req, err := a.TargetGroupDeploymentService.CreateTargetGroupDeployment(ctx, b)
```

Comment:
Any validation errors returned from this should be handled as a 400 response not a 500

Tests should validate this